src/appleseed/renderer/modeling/environmentedf/oslenvironmentedf.cpp

```diff
@@ -124,30 +240,75 @@ namespace
             Spectrum&               value,
             float&                  probability) const override
         {
-            const Vector3f local_outgoing = sample_sphere_uniform(s);
-            probability = RcpFourPi<float>();
 
+            if (!m_importance_sampler)
+            {
+                RENDERER_LOG_WARNING(
+                    "cannot sample osl environment edf \"%s\" because it is not bound to the environment.",
+                    get_path().c_str());
+                value.set(0.0f);
+                probability = 0.0f;
+                return;
+            }
+
+            // Sample the importance map.
+            size_t x, y;
+            Color3f payload;
+            float prob_xy;
+            m_importance_sampler->sample(s, x, y, payload, prob_xy);
+            assert(prob_xy >= 0.0f);
+
+            // Compute the coordinates in [0,1)^2 of the sample.
+            const float jitter_x = frac(s[0] * m_importance_map_width);
+            const float jitter_y = frac(s[1] * m_importance_map_height);
+            const float u = (x + jitter_x) * m_rcp_importance_map_width;
+            const float v = (y + jitter_y) * m_rcp_importance_map_height;
+            assert(u >= 0.0f && u < 1.0f);
+            assert(v >= 0.0f && v < 1.0f);
+
+            // Compute the spherical coordinates of the sample.
+            float theta, phi;
+            unit_square_to_angles(u, v, theta, phi);
+            shift_angles(theta, phi, m_theta_shift, m_phi_shift);
+
+            // Compute the local space emission direction.
+            const float cos_theta = cos(theta);
```

Comment:
Add `std::` qualifiers to all math functions (would also appreciate if you could fix the code you copied from).